### PR TITLE
Adds akka client, implemented as a persistent actor that keeps the position in the feed as internal state

### PR DESCRIPTION
**Pull request by JeroenVu**
Adds akka client, implemented as a persistent actor that keeps the position in the feed as internal state

### CODE REVIEW

File: modules/client-akka/src/main/scala/be/wegenenverkeer/atomium/client/AbstractAtomiumFeedClientActor.scala [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    case jsValue: JsValue => onJsValueEvent(jsValue)
+
+    case x: SnapshotOffer => positionInFeed = x.snapshot.asInstanceOf[Option[FeedPosition]]
+  }
+
+  def createEventToPublish(c: NewFeedEntryReceived[FEEDENTRY]): EVENT
+
+  def publishEvent(event: EVENT): Unit
+
+  def status(atomiumFeedClientStatus: AtomiumFeedClientStatus): Unit = ()
```
Comment (maesenka) [inline modules/client-akka/src/main/scala/be/wegenenverkeer/atomium/client/AbstractAtomiumFeedClientActor.scala]: What is the point of this method? Should document wat is the point of this method.
Also, should it not be abstract?

File: modules/client-akka/src/main/scala/be/wegenenverkeer/atomium/client/AbstractAtomiumFeedClientActor.scala [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+      subscribeToFeed()
+
+    case jsValue: JsValue => onJsValueEvent(jsValue)
+
+    case x: SnapshotOffer => positionInFeed = x.snapshot.asInstanceOf[Option[FeedPosition]]
+  }
+
+  def createEventToPublish(c: NewFeedEntryReceived[FEEDENTRY]): EVENT
+
+  def publishEvent(event: EVENT): Unit
```
Comment (maesenka) [inline modules/client-akka/src/main/scala/be/wegenenverkeer/atomium/client/AbstractAtomiumFeedClientActor.scala]: public abstract methods should be documented

File: modules/client-akka/src/main/scala/be/wegenenverkeer/atomium/client/AbstractAtomiumFeedClientActor.scala [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+            feedPosition = Some("beginOfFeed")
+          ))
+
+          feedObservableBuilder.inner.observeFromBeginning(config.pollingInterval.toMillis.toInt)
+
+      }
+      rx.lang.scala.JavaConversions.toScalaObservable(observable)
+    }
+
+    createObservable.subscribe(new Subscriber[FeedEntry[JsonNode]]() {
```
Comment (maesenka) [inline modules/client-akka/src/main/scala/be/wegenenverkeer/atomium/client/AbstractAtomiumFeedClientActor.scala]: Should this not be the FEEDENTRY type parameter?

File: modules/client-akka/src/main/scala/be/wegenenverkeer/atomium/client/AbstractAtomiumFeedClientActor.scala [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+
+  private def updateState(event: EVENT) = {
+    positionInFeed = Some(event.position)
+  }
+
+
+  protected def subscribeToFeed(): Unit = {
+
+    def createObservable: Observable[FeedEntry[JsonNode]] = {
+      val feedObservableBuilder = atomiumClient.feed(config.feedUrl, classOf[JsonNode])
```
Comment (maesenka) [inline modules/client-akka/src/main/scala/be/wegenenverkeer/atomium/client/AbstractAtomiumFeedClientActor.scala]: This now only works with Feeds using Json serialization

File: modules/client-akka/src/main/scala/be/wegenenverkeer/atomium/client/AbstractAtomiumFeedClientActor.scala [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+ *
+ * This actor only sends messages to itself. The feed entries are rapported by the [[publishEvent]] method.
+ *
+ * It's the job of the subclass to decide what to do with these entries.
+ *
+ * @param config feed and actor config
+ * @param eventFormat format for EventBus events
+ * @param feedEntryFormat Reads for reading entries in the feed
+ * @tparam EVENT EVENT type that will be put on the eventbus
+ * @tparam FEEDENTRY FEEDENTRY type in the feed
```
Comment (maesenka) [inline modules/client-akka/src/main/scala/be/wegenenverkeer/atomium/client/AbstractAtomiumFeedClientActor.scala]: FEEDENTRY is a confusing type parameter name here. It corresponds to the [T] parameter in FeedEntry[T] if I undertand correctly

File: modules/client-akka/src/main/scala/be/wegenenverkeer/atomium/client/AbstractAtomiumFeedClientActor.scala (hunk 4 above)
Comment (JeroenVu) [inline modules/client-akka/src/main/scala/be/wegenenverkeer/atomium/client/AbstractAtomiumFeedClientActor.scala]: The class documentation points out that it is a generic actor that consumes Atomium feeds _(in json format)_. I could rename the class to AbstractAtomiumJsonFeedClientActor?

File: modules/client-akka/src/main/scala/be/wegenenverkeer/atomium/client/AbstractAtomiumFeedClientActor.scala (hunk 5 above)
Comment (JeroenVu) [inline modules/client-akka/src/main/scala/be/wegenenverkeer/atomium/client/AbstractAtomiumFeedClientActor.scala]: It is indeed the type of a feed entry. What would you suggest as a less confusing type parameter name?

**Pull request merged**